src/python/pants/engine/legacy/graph.py

```diff
@@ -272,7 +288,7 @@ def _inject_address_specs(self, address_specs: AddressSpecs) -> Iterator[BuildFi
         logger.debug("Injecting address specs to %s: %s", self, address_specs)
         with self._resolve_context():
             (thts,) = self._scheduler.product_request(
-                LegacyTransitiveHydratedTargets, [address_specs]
+                LegacyTransitiveHydratedTargets, [Params(address_specs, self._options_bootstrapper)]
```

Comment:
So, the Options parsing that migrated into the engine is a bit fidgety due to singleton(s) that I didn't finish removing.

If it looks like this callsite is failing in odd ways in tests, you might need to mark that option value a RootRule and move to injecting only the relevant option as a `Param` here. But if not, don't worry about it!